hazelcast/src/test/java/com/hazelcast/internal/cluster/impl/MembershipFailureTest.java

```diff
@@ -350,9 +352,17 @@ public void slave_receives_member_list_from_non_master() {
         dropOperationsFrom(slave3, F_ID, singletonList(HEARTBEAT));
 
         suspectMember(slave2, master);
+        // workaround for timing issue introduced with reduction of join delays
```

Comment:
I'd rather that we find the issue than introduce workarounds, otherwise users upgrading to 4.1 (or other) will have to introduce such workarounds as well. Also, by merging this, we're introducing a change which either uncovers a bug or introduces a bug, and we're not exactly sure what's going on.

Reply:
Agreed. I just want to prove that just by putting delays in a couple of tests, the whole suite can pass, which was not happening for a wile before. I can gladly remove these workarounds and leave the PR as is to sit there while the root cause is investigated.

Reply:
Removed workarounds.